File: Output/output_Tabnine/python/Medium/1124-maximum-of-absolute-value-expression.py
#https://leetcode.com/problems/maximum-of-absolute-value-expression/
#Given two arrays of integers with equal lengths, return the maximum value of:
#|arr1[i] - arr1[j]| + |arr2[i] - arr2[j]| + |i - j|
#where the maximum is taken over all 0 <= i, j < arr1.length.
# 
#Example 1:
#Input: arr1 = [1,2,3,4], arr2 = [-1,4,5,6]
#Output: 13
#Example 2:
#Input: arr1 = [1,-2,-5,0,10], arr2 = [0,-2,-1,-7,-4]
#Output: 20
# 
#Constraints:
#	2 <= arr1.length == arr2.length <= 40000
#	-10^6 <= arr1[i], arr2[i] <= 10^6
class Solution(object):
    def maxAbsValExpr(self, arr1, arr2):
        """
        :type arr1: List[int]
        :type arr2: List[int]
        :rtype: int
        """
        n = len(arr1)
        max_val = float('-inf')
        
        # Iterate through all possible pairs of indices (i, j)
        for i in range(n):
            for j in range(i+1, n):
                # Calculate the absolute values and sum them up
                val = abs(arr1[i] - arr1[j]) + abs(arr2[i] - arr2[j]) + abs(i - j)
                
                # Update the maximum value found so far
                max_val = max(max_val, val)
        
        return max_val
        

File: Output/output_Tabnine/javascript/Medium/1124-maximum-of-absolute-value-expression.js
/**https://leetcode.com/problems/maximum-of-absolute-value-expression/ */
//Given two arrays of integers with equal lengths, return the maximum value of:
//|arr1[i] - arr1[j]| + |arr2[i] - arr2[j]| + |i - j|
//where the maximum is taken over all 0 <= i, j < arr1.length.
// 
//Example 1:
//Input: arr1 = [1,2,3,4], arr2 = [-1,4,5,6]
//Output: 13
//Example 2:
//Input: arr1 = [1,-2,-5,0,10], arr2 = [0,-2,-1,-7,-4]
//Output: 20
// 
//Constraints:
//	2 <= arr1.length == arr2.length <= 40000
//	-10^6 <= arr1[i], arr2[i] <= 10^6
/**
 * @param {number[]} arr1
 * @param {number[]} arr2
 * @return {number}
 */
var maxAbsValExpr = function(arr1, arr2) {
    let n = arr1.length;
    let maxVal = 0;
    
    for (let i = 0; i < n; i++) {
        for (let j = 0; j < n; j++) {
            let diff1 = Math.abs(arr1[i] - arr1[j]);
            let diff2 = Math.abs(arr2[i] - arr2[j]);
            let diff3 = Math.abs(i - j);
            maxVal = Math.max(maxVal, diff1 + diff2 + diff3);
        }
    }
    
    return maxVal;
    
};